base/strings/util.jl

```diff
@@ -536,7 +536,15 @@ julia> rtrunc("foo", 3)
 
 See also [`ltrunc`](@ref) and [`ctrunc`](@ref).
 """
-rtrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…') = struncate(str, maxwidth, replacement, :right)
+function rtrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…')
+    ret = string_truncate_boundaries(str, maxwidth, replacement, Val(:right))
+    if isnothing(ret)
+        return str
```

Comment:
```suggestion
        return string(str)
```
for type stability if `str` is not a `String`, to (hopefully?) match `str[begin:left] * replacement`.

Reply:
(Note that if `string(::String)` does not make a copy, so this won't slow things down in the common case of `String` arguments.)